### DISCUSSION TITLE
Function that transforms tw string to CSS rules object?

### DISCUSSION THREAD

**Discussion by piscopancer** [tailwindlabs/tailwindcss#13768] [Help] [answered]
I want to use classes inside my config like so
```ts
plugins: [
  plugin(({ addUtilities }) => {
    addUtilities({
      '.fancy': twToObject('bg-red-100 text-red-500 font-semibold'),
    })
  }),
],
```
So I need a function like above. I wonder if tailwind has such function internally which i can use in my code too

**Comment by wongjn** [comment, marked as answer]
You could use `@apply` as you would in CSS:

```js
plugins: [
  plugin(({ addUtilities }) => {
    addUtilities({
      '.fancy': {
         '@apply bg-red-100 text-red-500 font-semibold': {},
      },
    })
  }),
],
```
Though Adam Wathan, the creator of Tailwind, does not recommend this: https://github.com/tailwindlabs/tailwindcss/discussions/11012#discussioncomment-5653785.

**Comment by piscopancer** [comment]
@wongjn father of tailwind may not recommend it, but still i see it as a way to go, thanks!
